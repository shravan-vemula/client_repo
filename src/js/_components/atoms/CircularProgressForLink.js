import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from '@material-ui/core/CircularProgress';
import Box from '@material-ui/core/Box';
import { makeStyles } from "@material-ui/core/styles";
import COLORS from 'js/utils/colors';

const useStyles = makeStyles((theme) => ({

  circle : {
   color:COLORS.green
  },


}))

function CircularProgressWithLabel(props) {
  const classes = useStyles();
  return (
    <Box position="relative"
    height={170}
    width={170}
    display="flex"
    className={classes.circle}
     display="inline-flex">
      <CircularProgress className={classes.circle} variant="static" {...props} />
      </Box>
   
  );
}

CircularProgressWithLabel.propTypes = {
  /**
   * The value of the progress indicator for the determinate and static variants.
   * Value between 0 and 100.
   */
  value: PropTypes.number.isRequired,
};

export default function CircularProgressForLink() {
  const [progress, setProgress] = React.useState(30);

  React.useEffect(() => {
    const timer = setInterval(() => {
      setProgress((prevProgress) => (prevProgress >= 96 ? 100 : prevProgress + 33));
      
    }, 1600);
    
    return () => {
      clearInterval(timer);
    };
  }, []);

  return <CircularProgressWithLabel size={200} value={progress} />;
}