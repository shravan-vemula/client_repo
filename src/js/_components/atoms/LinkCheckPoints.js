import React from 'react';
import { makeStyles } from "@material-ui/core/styles";
import CircularProgress from '@material-ui/core/CircularProgress';

  const useStyles = makeStyles((theme) => ({
    root: {
      display: 'flex',
      '& > * + *': {
        marginLeft: theme.spacing(2),
      },
    },
    LinkText : {
        display:"inline-block",
        float:"right",
        textAlign:"left",
        position: "relative",
        top: "6%"
    },
    circleSize:{
        width:"20px !important",
        height:"8px !important",
        color:"green"
    },

    checkpointText:{
        marginLeft:"8px",
    }
  }));
  


function LinkCheckPoints() {
    const classes = useStyles();

    const [secureProgress, setSecureProgress] = React.useState(10);
    const [signinProgress, setSigninProgress] = React.useState(0);
    const [downloadProgress, setDownloadProgress] = React.useState(0);
  

    

    React.useEffect(() => {
    const timer1 = setInterval(() => {
        setSecureProgress((prevSecureProgress) => (prevSecureProgress >= 100 ? 100: prevSecureProgress + 50));
      }, 160);
  
      const timer2 = setInterval(() => {
        setSigninProgress((prevSigninProgress) => (prevSigninProgress >= 100 ? 100 : prevSigninProgress + 20));
        
      }, 320);
  
      const timer3 = setInterval(() => {
        setDownloadProgress((prevDownloadProgress) => (prevDownloadProgress >= 90 ? 100 : prevDownloadProgress + 30));
       
      }, 640);
  
    
      return () => {
        clearInterval(timer1);
        clearInterval(timer2);
        clearInterval(timer3);
      };
    }, []);


    return (
        <div id="checkpoints-text" className={classes.LinkText}  style={{fontFamily:"Inter,sans-serif"}}>
            <div><CircularProgress className={classes.circleSize} variant="static" value={secureProgress} />   <span className={classes.checkpointText}>Securing Connection</span></div> 
                <br />
                <br />
                <div><CircularProgress className={classes.circleSize} variant="static" value={signinProgress} /> <span className={classes.checkpointText}> Signing in</span></div>
                <br />
                <br />
                <div><CircularProgress className={classes.circleSize} variant="static" value={downloadProgress} /> <span className={classes.checkpointText}> Downloading Transactions</span> </div>

        </div>
    )
}

export default LinkCheckPoints
