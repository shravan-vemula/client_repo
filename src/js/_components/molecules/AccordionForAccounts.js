import React,{useState,useEffect} from 'react';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import MuiAccordion from '@material-ui/core/Accordion';
import MuiAccordionSummary from '@material-ui/core/AccordionSummary';
import MuiAccordionDetails from '@material-ui/core/AccordionDetails';
import AccountBalanceIcon from '@material-ui/icons/AccountBalance';
import AccountBalanceWalletIcon from '@material-ui/icons/AccountBalanceWallet';
import PaymentIcon from '@material-ui/icons/Payment';
import MoneyIcon from '@material-ui/icons/Money';
import COLORS from "js/utils/colors";


const Accordion = withStyles({
  root: {
    border: '1px solid rgba(0, 0, 0, .125)',
    boxShadow: 'none',
    '&:not(:last-child)': {
      borderBottom: 0,
    },
    '&:before': {
      display: 'none',
    },
    '&$expanded': {
      margin: 'auto',
    },
  },
  expanded: {},
})(MuiAccordion);

const AccordionSummary = withStyles({
  root: {
    backgroundColor: 'rgba(0, 0, 0, .03)',
    borderBottom: '1px solid rgba(0, 0, 0, .125)',
    marginBottom: -1,
    minHeight: 56,
    '&$expanded': {
      minHeight: 56,
    },
  },
  content: {
    '&$expanded': {
      margin: '12px 0',
    },
  },
  expanded: {},
})(MuiAccordionSummary);

const AccordionDetails = withStyles((theme) => ({
  root: {
    padding: theme.spacing(2),
  },
}))(MuiAccordionDetails);




const useStyles = makeStyles((theme)=>({

    amountText:{
        width: "100%"
    },

    category:{
      position:"relative",
      bottom:"22%",
      marginLeft:"10px",
      color:COLORS.default
    },


    amountInHand : {
      color:COLORS.green,
        display:"inline-block",
        float : "right"
    },

    amountSpent : {
        color:COLORS.red,
        display:"inline-block",
        float : "right"
    },

    amount : {
     
      display:"inline-block",
      float : "right"
  },

  disabledBackground:{
      backgroundColor:COLORS.white + "!important"
  }

}))


export default function AccordionForAccounts() {

    const [transactionList , setTransactionsList] = useState([])
    const [budgetList , setBudgetList] = useState([])


    useEffect(() => {
      fetch("https://bc3-json-mock.herokuapp.com/expenses", {
        method: "GET",
      })
        .then((response) => {
          response.json().then((result) => {
            setTransactionsList(result);
          });
        })
        .catch((err) => {});

      fetch("https://bc3-json-mock.herokuapp.com/budgetComponents", {
        method: "GET",
      })
        .then((response) => {
          response.json().then((result) => {
            setBudgetList(result);
          });
        })
        .catch((err) => {});
  
    }, [])

    



 
  const classes = useStyles();

  console.log(transactionList)
  let expenseAmount = 0
  transactionList.filter((transaction)=>transaction.type==="debit").map((transaction) => (expenseAmount = expenseAmount + (transaction.amount * 1)));

  let budgetAmount = 0
  budgetList.map((transaction) => (budgetAmount = budgetAmount + (transaction.currency * 1)));
 let incomeValue=0
 transactionList.filter((transaction)=>transaction.type==="credit").map((transaction) => (incomeValue = incomeValue + (transaction.amount * 1)));
  return (
    <div>
      <Accordion  >
        <AccordionSummary aria-controls="panel1d-content" id="panel1d-header">
           <div className={classes.amountText}><AccountBalanceIcon/><span className={classes.category}> Cash in Accounts</span><div id="amount" className={classes.amountInHand}>{incomeValue}</div></div>
        </AccordionSummary>
         <AccordionDetails>          
  <div className={classes.amountText}>Axis Bank<div id="amount" className={classes.amountInHand}>{incomeValue/2}</div></div>          
            </AccordionDetails>
       
        <AccordionDetails>          
            <div className={classes.amountText}>HDFC <div id="amount" className={classes.amountInHand}>{incomeValue/2}</div></div>          
        </AccordionDetails>
      </Accordion>
      <Accordion >
        <AccordionSummary aria-controls="panel2d-content" id="panel2d-header" data-testId="AccordionSummaryBudget">
  <div className={classes.amountText}><AccountBalanceWalletIcon /><span className={classes.category}>Budget</span> <div id="amount" className={classes.amount}>{budgetAmount}</div></div>
        </AccordionSummary>
        {budgetList.map((metrics, index ) => {
            index += 1;
            return(
        <AccordionDetails data-testId="budgetData">          
            <div  className={classes.amountText}>{metrics.category} <div id="amount" className={classes.amount}>{metrics.currency }</div></div>          
        </AccordionDetails>
            )
          })}
      </Accordion>
      <Accordion >
        <AccordionSummary aria-controls="panel3d-content" id="panel3d-header" data-testId="expensesBar">
      <div className={classes.amountText}><PaymentIcon /><span className={classes.category}>Expenses</span> <div id="amount" className={classes.amountSpent}>{expenseAmount}</div></div>
        </AccordionSummary>
        {transactionList.map((metrics, index ) => {
            index += 1;
            return(
        <AccordionDetails>          
            <div className={classes.amountText}>{metrics.category} <div id="amount" className={classes.amountSpent}>{metrics.amount }</div></div>          
        </AccordionDetails>
            )
          })}
      </Accordion>
      <Accordion disabled className={classes.disabledBackground}>
        <AccordionSummary aria-controls="panel3d-content" id="panel3d-header">
        <div className={classes.amountText}><MoneyIcon /><span className={classes.category}>Balance amount in budget</span> <div id="amount" className={classes.amount}>{budgetAmount - expenseAmount}</div></div>
        </AccordionSummary>
      </Accordion>
    </div>
  );
}