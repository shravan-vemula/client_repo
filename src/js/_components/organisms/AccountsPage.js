import React from "react";
import { makeStyles, Grid, Paper } from "@material-ui/core";
import AccordionForAccounts from "../molecules/AccordionForAccounts";
import { myTheme } from "theme";

const useStyles = makeStyles((theme) => ({
  accountsDisplay: {
    width: "80%",
    marginLeft: "10%",
    marginTop: "5%",
    fontFamily: myTheme.typography.fontFamily.default + "!important",
  },
  titleDisplay: {
    fontSize: "30px",
    fontStyle: "normal",
    fontFamily: "Inter,sans-serif",
    fontWeight: "normal",
    lineHeight: "1.47",
    fontStretch: "normal",
    letterSpacing: "normal",
  },
}));

function AccountsPage() {
  const classes = useStyles(myTheme);
  return (
    <>
      <div id="account-overiview" className={classes.accountsDisplay}>
        <Grid container spacing={3}>
          <Grid item xs={12} sm={12} md={12}>
            <Paper elevation={0}>
              {" "}
              <h1 className={classes.titleDisplay}>Accounts</h1>
            </Paper>
          </Grid>
          <Grid item xs={10} sm={10} md={10}>
            <AccordionForAccounts />
          </Grid>
        </Grid>
      </div>
    </>
  );
}

export default AccountsPage;
