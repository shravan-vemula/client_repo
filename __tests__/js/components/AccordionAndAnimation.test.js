import React from 'react';
import AccordionForAccounts from 'js/_components/molecules/AccordionForAccounts';
import { render, screen, fireEvent } from '@testing-library/react';
import {BrowserRouter as Router, MemoryRouter} from 'react-router-dom';
import CircularProgressForLink from 'js/_components/atoms/CircularProgressForLink';
import LinkCheckPoints from 'js/_components/atoms/LinkCheckPoints';
import AnimantionModalSuccess from 'js/_components/atoms/AnimationModalSuccess';
import AccountsPage from 'js/_components/organisms/AccountsPage';



const fetch = jest.fn();
global.fetch = fetch;

beforeEach(() => {
  fetch.mockClear();

  fetch.mockResolvedValue({
    key: "value", // return value on fetch resolve
  });
});


describe('Acoordion renders correctly', () =>{

    test('renders AccountsPage', () =>{
        render(<Router><AccountsPage /></Router>);
        expect(screen.getAllByText("Accounts")).toBe;
    })  



    it("Accordion snapshot", () => {
        const wrapper = shallow(<Router><AccordionForAccounts /> </Router>);
        expect(wrapper).toMatchSnapshot();
      });


    test('renders accordion', () =>{
        render(<Router><AccordionForAccounts /></Router>);
        expect(screen.getByText('Balance amount in budget')).toBe;
        screen.debug();
    })    

    it("Accordion click",() =>{
        const { getByTestId } =  render(<Router><AccordionForAccounts /></Router>);
        expect(getByTestId("AccordionSummaryBudget")).toBe;
        fireEvent.click(getByTestId("AccordionSummaryBudget"), {
        });
        
    });

    test('renders AnimantionModalSuccess and click goToOverView', () =>{
        const { getByTestId } = render(<Router><AnimantionModalSuccess placeHolder={"Enter Andhra Bank Id"}/></Router>);
        expect(getByTestId("goToOverView")).toBe
        fireEvent.click(getByTestId("goToOverView"))

    })  

    test('renders AnimantionModalSuccess and click  ', () =>{
        const { getByTestId } = render(<Router><AnimantionModalSuccess placeHolder={"Enter Andhra Bank Id"}/></Router>);
        expect(getByTestId("addAnotherAccount")).toBe
        fireEvent.click(getByTestId("addAnotherAccount"))
    })  
    
    test('renders LinkCheckPoints', () =>{
        render(<Router><LinkCheckPoints /></Router>);

    })  

    it("Snapshot CircularProgressForLink", () => {
        const wrapper = shallow(<Router><CircularProgressForLink /> </Router>);
        expect(wrapper).toMatchSnapshot();
      });

    it("SnapSHot LinkCheckPoints", () => {
        const wrapper = shallow(<Router><LinkCheckPoints /> </Router>);
        expect(wrapper).toMatchSnapshot();
      });



});
