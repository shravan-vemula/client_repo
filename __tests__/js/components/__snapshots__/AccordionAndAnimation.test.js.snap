// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Acoordion renders correctly Accordion snapshot 1`] = `ShallowWrapper {}`;

exports[`Acoordion renders correctly SnapSHot LinkCheckPoints 1`] = `ShallowWrapper {}`;

exports[`Acoordion renders correctly Snapshot CircularProgressForLink 1`] = `ShallowWrapper {}`;
